src/liberty/devCommands.ts

```diff
@@ -610,4 +612,44 @@ async function getLocalGradleWrapper(projectFolder: string): Promise<string | un
  */
 function isWin(): boolean {
     return process.platform.startsWith("win");
-}
\ No newline at end of file
+}
+
+/*
+will return the path of the report, since there are diffrent folders to look into and the file names can be different 
+we need to get the paths to look for dynamically
+*/
+function getReportFile(path : any, dir : string, filename : string) : any{
+    return Path.join(path, "target", dir, filename);
+}
+
+/*
+Function will check if the report is available within the given path and returns a boolean based on it and also 
+  the report will be displayed if it is available
+*/
+function checkReportAndDisplay(report : any, reportType : string, reportTypeLabel : string, libProject : LibertyProject, directory : string) : Promise<boolean> {
+    return new Promise((resolve) => {
+      fs.exists(report, (exists) => {
+        if(exists){
+            const panel = vscode.window.createWebviewPanel(
+                reportType, // Identifies the type of the webview. Used internally
+                libProject.getLabel() + " " + reportTypeLabel + " report", // Title of the panel displayed to the user
+                vscode.ViewColumn.Two, // Open the panel in the second window
+                {}, // Webview options
+            );
+            panel.webview.html = getReport(report); // display HTML content
+            /*
+            For maven projects we need to check for the report in 'reports' and 'site', we only need to show the message if it is not 
+            available in both the locations, below condition make sure to avoid the message when its a maven project and the directory 
+            is 'reports'
+            */ 
+        }else if(!((libProject.getContextValue() === LIBERTY_MAVEN_PROJECT || libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER) 
+                    && directory=="reports")){
```

Comment:
This `if` statement may not always apply. If the directory changes again or the order we check the files changes, or if gradle projects become involved, we have to reconsider the logic. Since the logic to know if we want the error message to be displayed or not is determined outside of this method, it might be better to have a boolean variable named `showErrorMessage` that is passed in when this method is called.